src/mkdocstrings/handlers/base.py

```diff
@@ -232,7 +232,10 @@ def update_env(self, md: Markdown, config: dict) -> None:
         # Prevent a bug that happens due to treeprocessors running on the same fragment both as the inner doc and as
         # part of the re-integrated doc. Namely, the permalink '¶' would be appended twice. This is the only known
         # non-idempotent effect of an extension, so specifically prevent it on the inner doc.
-        configs.setdefault("toc", {})["permalink"] = False
+        try:
+            configs["toc"] = dict(configs["toc"], permalink=False)
+        except KeyError:
+            pass
```

Comment:
Shouldn't we simply deep copy `configs` at line 231 (before the comment)? If we find other non-idempotent extensions, we'll have to write a try except for each, no? I find the deep copy easier to reason about and less prone to future errors, what do you think?

Reply:
deepcopy is too big of a hammer. You never know what things will end up being there. I think this is clearly better for a stable release. And long-term, this will not be a thing. I'm dropping all of this code here https://github.com/pawamoy/mkdocstrings/pull/209/files#diff-288aa0bb4705e3681add5ce91b76e798bcf7b7ba2570563ec6e77809fa2bc166L232  (and the reason that I noticed this regression is that that refactor avoids this regression :smile: )

Reply:
Oh, OK, makes sense, thanks